File: .tmp_py4013/batch_09_dump/3018_maximum_number_of_removal_queries_that_can_be_processed_i.js
// LeetCode 3018 - Maximum Number of Removal Queries That Can Be Processed I
// https://leetcode.com/problems/maximum-number-of-removal-queries-that-can-be-processed-i/

var maximumProcessableQueries = function(nums, queries) {
    const n = nums.length;
    const f = Array.from({length: n}, () => new Array(n).fill(0));
    const m = queries.length;
    for (let i = 0; i < n; i++) {
        for (let j = n - 1; j >= i; j--) {
            if (i > 0) {
                const t = f[i - 1][j] < m && nums[i - 1] >= queries[f[i - 1][j]] ? 1 : 0;
                f[i][j] = Math.max(f[i][j], f[i - 1][j] + t);
            }
            if (j + 1 < n) {
                const t = f[i][j + 1] < m && nums[j + 1] >= queries[f[i][j + 1]] ? 1 : 0;
                f[i][j] = Math.max(f[i][j], f[i][j + 1] + t);
            }
            if (f[i][j] === m) return m;
        }
    }
    let ans = 0;
    for (let i = 0; i < n; i++) {
        const t = f[i][i] < m && nums[i] >= queries[f[i][i]] ? 1 : 0;
        ans = Math.max(ans, f[i][i] + t);
    }
    return ans;
};

File: .tmp_rs4013/rust_check13/3018_maximum_number_of_removal_queries_that_can_be_processed_i.rs
#![allow(dead_code, unused_variables, unused_mut, unused_assignments, unused_imports)]
struct Solution;
// LeetCode 3018 - Maximum Number of Removal Queries That Can Be Processed I
// https://leetcode.com/problems/maximum-number-of-removal-queries-that-can-be-processed-i/

impl Solution {
    pub fn maximum_processable_queries(nums: Vec<i32>, queries: Vec<i32>) -> i32 {
        let n = nums.len();
        let mut f = vec![vec![0; n]; n];
        let m = queries.len();
        for i in 0..n {
            for j in (i..n).rev() {
                if i > 0 {
                    let qi = f[i - 1][j] as usize;
                    if qi >= m {
                        return m as i32;
                    }
                    let t = if nums[i - 1] >= queries[qi] { 1 } else { 0 };
                    f[i][j] = f[i][j].max(f[i - 1][j] + t);
                }
                if j + 1 < n {
                    let qi = f[i][j + 1] as usize;
                    if qi >= m {
                        return m as i32;
                    }
                    let t = if nums[j + 1] >= queries[qi] { 1 } else { 0 };
                    f[i][j] = f[i][j].max(f[i][j + 1] + t);
                }
                if f[i][j] == m as i32 {
                    return m as i32;
                }
            }
        }
        let mut ans = 0;
        for i in 0..n {
            let qi = f[i][i] as usize;
            if qi >= m {
                return m as i32;
            }
            let t = if nums[i] >= queries[qi] { 1 } else { 0 };
            ans = ans.max(f[i][i] + t);
        }
        ans
    }
}
